src/main/java/decodes/cwms/CwmsTimeSeriesDb.java

```diff
@@ -2309,4 +1910,95 @@ public String flags2display(int flags)
 		return CwmsFlags.flags2Display(flags);
 	}
 
+	@Override
+	public Connection getConnection()
+	{
+		// Called from DAOs to get a new connection from the pool.
+		if (conInfo == null || conInfo.getLoginInfo() == null)
+		{
+			failure("CwmsTimeSeriesDb.getConnection -- loginInfo is null! DB not initialized?");
+			// TODO should throw BadConnectException?
+			return null;
+		}
+		
+		Connection ret = null;
+		try
+		{
+			ret = CwmsDbConnectionPool.getInstance().getConnection(conInfo.getLoginInfo());
```

Comment:
I recommend adding the second String parameter in order to setup the end-to-end metrics.

Reply:
Done.